SIT.Manager.Avalonia/ViewModels/PlayPageViewModel.cs

```diff
@@ -251,35 +230,39 @@ private async Task ConnectToServer(bool launchServer = false)
                 }
             }
 
-            if (launchServer)
-            {
-                using CancellationTokenSource cts = new();
+            if (launchServer) {
                 _akiServerService.Start();
 
-                DateTime abortTime = DateTime.Now + TimeSpan.FromSeconds(30);
-                cts.CancelAfter(abortTime - DateTime.Now);
-                cts.Token.Register(() =>
-                {
-                    Dispatcher.UIThread.InvokeAsync(() =>
-                    {
-                        new ContentDialog()
-                        {
-                            Title = "Server Error",
-                            Content = "The server never started. Please check the logs for more information.",
-                            CloseButtonText = "Ok"
-                        }.ShowAsync();
-                    });
-                });
-
-                TimeSpan timeToAbort = default;
-                while ((timeToAbort = abortTime - DateTime.Now).TotalSeconds > 0 && !cts.IsCancellationRequested && !_akiServerService.IsStarted)
-                {
-                    QuickPlayText = $"Waiting for server ({timeToAbort.TotalSeconds:N0}s)";
-                    await Task.Delay(500);
+                bool aborted = false;
+                RunningState serverState = _akiServerService.State;
+                while (serverState == RunningState.Starting) {
```

Comment:
What do you think about moving the assignment to the condition? e.g `(serverState = _akiServerService.State) == RunningState.Starting`